src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -93,4 +99,139 @@ public static String toQueryString(Map<String, String> params) {
         result.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
         return result;
     }
+
+    /**
+     * TODO: fully implement
+     */
+    public static boolean itineraryIsSavable(Itinerary itinerary) {
+        return true;
+    }
+
+    /**
+     * Returns true if the itineraries match for the purposes of trip monitoring.
+     *
+     * @param previousItinerary The original itinerary that others are compared against.
+     * @param newItinerary A new itinerary that might match the previous itinerary.
+     */
+    public static boolean itinerariesMatch(Itinerary previousItinerary, Itinerary newItinerary) {
+        // make sure either itinerary is monitorable before continuing
+        if (!itineraryIsSavable(previousItinerary) || !itineraryIsSavable(newItinerary)) return false;
+
+        // make sure itineraries have same amount of legs
+        if (previousItinerary.legs.size() != newItinerary.legs.size()) return false;
+
+        // make sure each leg matches
+        for (int i = 0; i < previousItinerary.legs.size(); i++) {
+            Leg previousItineraryLeg = previousItinerary.legs.get(i);
+            Leg newItineraryLeg = newItinerary.legs.get(i);
+
+            if (!legsMatch(previousItineraryLeg, newItineraryLeg)) return false;
+        }
+
+        // if this point is reached, the itineraries are assumed to match
+        return true;
+    }
+
+    /**
+     * Check whether a new leg of an itinerary matches the previous itinerary leg for the purposes of trip monitoring.
+     */
+    private static boolean legsMatch(Leg previousItineraryLeg, Leg newItineraryLeg) {
+        // for now don't analyze non-transit legs
+        if (!previousItineraryLeg.transitLeg) return true;
+
+        // make sure the same from/to stop are being used
+        if (
+            !stopsMatch(previousItineraryLeg.from, newItineraryLeg.from) ||
+                !stopsMatch(previousItineraryLeg.to, newItineraryLeg.to)
+        ) {
+            return false;
+        }
+
+        // make sure the transit service is the same as perceived by the customer
```

Comment:
> Make sure the transit service is the same as perceived by the customer. In other words, verify that mode, route/agency names, and other leg details remain unchanged.

Reply:
Fixed in ffc3b49b3ad576d936b04758d572efdf3f217ab2